### ISSUE TITLE
[FEA] select function for columns and scalars

### ISSUE THREAD

**Issue by jlowe** [rapidsai/cudf#3042]
**Is your feature request related to a problem? Please describe.**
A SQL `CASE` statement is like an 'if/else'.  For example:
```sql
CASE WHEN (x > 20) THEN x - 20 ELSE x + y + 2
```
One approach is to generate the answers for both expressions then select the answer for the final result based on the condition.  A select function would be useful in implementing that approach.

**Describe the solution you'd like**
libcudf would provide a function that is called something like this:
```c++
  col_out = cudf::select(col_a, col_b, col_c);
```
where `col_a` and `col_b` are columns of the same type and `col_c` is a `bool8` column.  `col_c` would typically be generated as the result of a comparison operation.

The `select` function would compute the result of this expression for every row:
```c++
col_out[row_index] = (col_c[row_index]] == 0) ? col_a[row_index] : col_b[row_index];
```

Ideally the `select` function could also operate on scalar inputs interchangeably with columns.   That would be useful when the `THEN` and/or `ELSE` expressions are not columnar (e.g.: a literal scalar) to avoid needing to fill a column with the scalar before calling `select`.

**Describe alternatives you've considered**
cudf's `transform` could apply if there's only one input column to the conditional expression.  However it cannot be used beyond the unary case.

Ideally a single kernel could evaluate an expression tree directly on all input columns and generate the output column.  That would be more efficient than generating all three expression columns (`WHEN`, `THEN`, and `ELSE`) before generating the output.  Implementing that is significant work.  `select`  would be easier to implement and useful until a more sophisticated and efficient approach is developed.

**Comment by jrhemstad** [comment]
I believe that https://github.com/rapidsai/cudf/issues/2428 could be the solution to using a single kernel to evaluate the expressions directly and materialize the output. 

However, this would require the ability to generate either the corresponding C/PTX from Spark such that we can feed it into Jitify/NVRTC and compile the single kernel. I do not believe this functionality yet exists.

**Comment by harrism** [comment]
@jlowe is this fixed by adding `copy_if_else`?

**Comment by jlowe** [comment]
Yes, this can be implemented by `copy_if_else` introduced in #3331.
